airbyte-db/db-lib/src/main/java/io/airbyte/db/factory/ConnectionFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.factory;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * This class as been added in order to be able to save the connection in a test. It was found that
+ * the {@link javax.sql.DataSource} close method wasn't propagating the connection properly. It
+ * shouldn't be needed in our application code.
```

Comment:
What does 'shouldn't be needed in our application code' mean?

Should we replace DataSource through database code base with this?

Reply:
This factory, I mean that we can use a datasource directly. Since it is created using Micronaut, it will be a singleton and thus reduce the leak risk. Changing that in our applications will make use to do unnecessary changes knowing that the datasource is not even created by use and managed through the application.yaml. It will need us to re-implement some options that we are using.

Here is an example of a datasource declaration from the worker:
```
datasources:
  config:
    connection-test-query: SELECT 1
    connection-timeout: 30000
    idle-timeout: 600000
    initialization-fail-timeout: -1 # Disable fail fast checking to avoid issues due to other pods not being started in time
    maximum-pool-size: 20
    minimum-idle: 0
    url: ${DATABASE_URL}
    driverClassName: org.postgresql.Driver
    username: ${DATABASE_USER}
    password: ${DATABASE_PASSWORD}
    pool-name: config-pool
```

Reply:
In that case, can we clarify with something like, this does not apply to DI frameworks e.g. Micronaut?

'Application code' is confusing because there is a wide breadth of application code at Airbyte. A reader is going to get confused if they are going between Platform (Micronaut-ed) and Connectors (not-micronauted)